src/Domains/Event/Events/Actions/CreateEventAction.php

```diff
@@ -218,4 +228,35 @@ protected function storeEventResources(ModelsEvent $event, array $resources): vo
             }
         }
     }
+
+    protected function validateTimeSlotAvailability(): void
+    {
+        if (! $this->event->dates->count()) {
+            return; // No dates to validate
+        }
+
+        $dateData = $this->event->dates[0]; // Assuming single date for now
```

Comment:
@jesusantguerrero need to confirm it will always be a earray?

Reply:
yeah, I keept the previous structure here